Open_Weather-in-hook-and-redux/src/components/App.tsx

```diff
@@ -0,0 +1,88 @@
+import { Component } from "react"
+import { Weather } from "../types/Weather";
+import { Input, Info, DropDown } from './common/index';
+import { debounce } from "lodash";
+import css from './app.module.css';
+
+interface AppState {
+    searchCity: string;
+    unit: string;
+    weather: Weather;
+    
+}
+
+const myFetch = (url: string) => {
+    return fetch(url).then((data) => {
+      if (data.ok) {
+        return data.json();
+      }
+      throw Error("oops");
+    });
+  };
+
+export class App extends Component<{}, AppState> {
+    isOffline = true
+
+    state: AppState = {
+        searchCity: 'Minsk',
+        unit: "metric",
+        weather: { main: {
+          temp: 0,
+          feels_like: 0,
+          pressure: 0,
+          humidity: 0
+        }, 
+        weather: [{icon: undefined}],
+        name: '' },
+    }
+
+  
+
+  fetchWeatherDebounced = debounce(this.componentDidMount, 1_500);
```

Comment:
мы никогда сами не вызываем никакие методы жизненного цикла. Вместо это нужно создать функцию fetchWeather и использовать ее тут и во всех методах жизненного цикла (в didMount и didUpdate)
```suggestion
  fetchWeatherDebounced = debounce(this.fetchWeather, 1_500);
```